### DISCUSSION TITLE
simultaneous shutter button functionality is missing from xdrv_27_esp32_shutter.ino

### DISCUSSION THREAD

**Discussion by erwann2000** [arendst/Tasmota#19413] [Q&A] [answered]
Hello,
I have a sonoff dualr3 and shelly 2pm with a wall-mount shutter command with up,down and stop buttons with only 2 contacts. Stop button close 2 contacts simultaneously.

I tried long time to enable rule with Shutter1#Button0 to add stop command as explained in the Shutters and Blinds documentation.
Then I tried to search in code, I see xdrv_27_shutter.ino have the functionnality but not xdrv_27_esp32_shutter.ino

Can you help me ?

**Comment by erwann2000** [comment]
Hello @stefanbode,
Do you think this feature will be available in the future?

**Reply by stefanbode** [reply]
What is the specific use case behind this? As I understood you press up and down at the same time.

**Reply by stefanbode** [reply]
Ah I see above. Hidden stop button. Let me have a look next week. There was a reason to skip it

**Reply by erwann2000** [reply]
Exactly, this is Legrand button 077025. It pushes mechanically up and down during the stop action. At the end of the rule section of the Shutters documentation, I see the possibility of Simultaneous button press, https://tasmota.github.io/docs/Blinds-and-Shutters/#rules

**Reply by stefanbode** [reply]
Thanks for guiding me to the rules section. I was not aware of this. This code was added from another Tasmota friend. I will Check and use the rules for testing functions.

**Reply by stefanbode** [reply]
ok this is not an easy fix. with ESP32 I removed all the duplicate code that is also in the button_v4 driver and only listen to the FUNC_BUTTON_PRESSED code. The problem is that this does not know the SIMULTANOUS pressed. Also there is a trick that the BUTTON number changes to 0 and then you get a button0 = 1 or button0 = hold. I have to check how to initiate this on top. Will take some time

**Reply by stefanbode** [reply]
Submitted an enhancement: https://github.com/arendst/Tasmota/pull/19465

**Comment by stefanbode** [comment, marked as answer]
Fix submitted to reenable on esp32: https://github.com/arendst/Tasmota/pull/19465

**Reply by erwann2000** [reply]
thanks a lot @stefanbode , this works so fine
